feature-tests/utils/send-sqs-message.ts

```diff
@@ -32,3 +32,56 @@ function getCurrentTimestamp(date = new Date()): CurrentTimeDescriptor {
     seconds: Math.floor(date.valueOf() / 1000),
   };
 }
+
+export async function purgeEgressQueue() {
+  const sqs = new SQS({ apiVersion: '2012-11-05', region: process.env.AWS_REGION });
+  const queueURL = EndPoints.SQS_EGRESS_QUEUE_URL;
+  const parameters = {
+    QueueUrl: queueURL,
+  };
+  try {
+    await sqs.purgeQueue(parameters);
+    await timeDelayForTestEnvironment(5000);
+    console.log('Purge Success');
+  } catch (error) {
+    console.log('Error', error);
+  }
+}
+
+export async function receiveMessagesFromEgressOueue() {
+  const sqs = new SQS({ apiVersion: '2012-11-05', region: process.env.AWS_REGION });
+  let response;
+  const queueURL = EndPoints.SQS_EGRESS_QUEUE_URL;
+  const messages = [];
+
+  const parameters = {
+    QueueUrl: queueURL,
+    MaxNumberOfMessages: 10,
+  };
+  let count = 0;
+  do {
+    try {
+      response = await sqs.receiveMessage(parameters);
+
+      if (response?.Messages) {
+        for (const message of response.Messages) {
+          messages.push(message);
+        }
+      }
+    } catch (error) {
+      console.log('No messages in Queue', error);
+    }
+    count += 1;
+    if (count > 10) break;
+  } while (response?.Messages && response.Messages.length > 0);
+  return messages;
+}
+
+export async function filterUserIdInMessages(testUserId: string) {
+  const messages = await receiveMessagesFromEgressOueue();
+  const filteredMessageByUserId = messages.filter((message) => {
+    const messageBody = message.Body ? JSON.parse(message.Body) : {};
```

Comment:
`JSON.parse` could throw an error, so might be worth wrapping it in a try catch block?